compio-net/src/socket.rs

```diff
@@ -128,6 +128,10 @@ impl Socket {
 
     pub async fn bind(addr: &SockAddr, ty: Type, protocol: Option<Protocol>) -> io::Result<Self> {
         let socket = Self::new(addr.domain(), ty, protocol).await?;
+        #[cfg(not(windows))]
+        if let Some(Protocol::TCP) = protocol {
```

Comment:
I suggest changing this method to `bind(addr, ty, protocol, reuseaddr: bool)` to avoid matching protocol here.

Reply:
Per discussions in telegram I will change `TcpListener` instead.